Source/Common/Config.h

```diff
@@ -19,11 +19,33 @@ namespace FEX::Config {
 
   void SaveLayerToJSON(const std::string& Filename, FEXCore::Config::Layer *const Layer);
 
-  std::pair<std::string, std::string> LoadConfig(
+  struct ApplicationNames {
+    // This is the full path to the program (if it exists).
+    std::string ProgramPath;
+    // This is the program executable name (if it exists).
+    std::string ProgramName;
+  };
+
+  /**
+   * @brief Loads the FEX and application configurations for the application that is getting ready to run.
+   *
+   * @param NoFEXArguments Do we want to parse FEXLoader arguments, Or is this FEXInterpreter?
+   * @param LoadProgramConfig Do we want to load application specific configurations?
+   * @param argc The `argc` passed to main(...)
+   * @param argv The `argv` passed to main(...)
+   * @param envp The `envp` passed to main(...)
+   * @param ExecFDInterp If FEX was executed with binfmt_misc FD argument
+   * @param ProgramFDFromEnv If FEX was executed with execveat FD argument
+   *
+   * @return The application name and path structure
+   */
+  ApplicationNames LoadConfig(
     bool NoFEXArguments,
     bool LoadProgramConfig,
     int argc,
     char **argv,
-    char **const envp
+    char **const envp,
+    bool ExecFDInterp,
+    bool ProgramFDFromEnv
```

Comment:
Since the we're only passing `!!getenv(...)` (and false) in here anyway, how about passing in the full value as a `string_view` or an `optional` instead?

That reduces the amount of hidden state accessed in LoadConfig, avoids env-querying twice, and removes one of the weakly typed boolean toggles in the parameter list.

Reply:
Switched over to a string_view.